pkg/controller/job/job_controller_test.go

```diff
@@ -5172,6 +5198,54 @@ func TestFinalizersRemovedExpectations(t *testing.T) {
 	}
 }
 
+func TestBackupFinalizers(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	clientset := fake.NewSimpleClientset()
+	sharedInformers := informers.NewSharedInformerFactory(clientset, controller.NoResyncPeriodFunc())
+	manager := NewController(ctx, sharedInformers.Core().V1().Pods(), sharedInformers.Batch().V1().Jobs(), clientset)
+	manager.podStoreSynced = alwaysReady
+	manager.jobStoreSynced = alwaysReady
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	podInformer := sharedInformers.Core().V1().Pods().Informer()
+	go podInformer.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, podInformer.HasSynced)
+
+	// 1. Create the controller but do not start the workers.
+	// This is done above by initializing the manager and not calling manager.Run() yet.
+
+	// 2. Create a job.
+	job := newJob(2, 2, 6, batch.NonIndexedCompletion)
+
+	// 3. Create the pods.
+	podBuilder := buildPod().name("test_pod").deletionTimestamp().trackingFinalizer().job(job)
+	pod, err := clientset.CoreV1().Pods("default").Create(context.Background(), podBuilder.Pod, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Creating pod: %v", err)
+	}
+
+	// 4. Finish the job.
+	job.Status.Conditions = append(job.Status.Conditions, batch.JobCondition{
+		Type:   batch.JobComplete,
+		Status: v1.ConditionTrue,
+	})
+
+	// 5. Start the workers.
+	go manager.Run(context.TODO(), 1)
```

Comment:
```suggestion
	go manager.Run(context.TODO(), 0)
```

To be 100% that this is the backup reconciler removing the finalizers, not the main job reconciler.

Reply:
Ah! Maybe this is why there is no coverage.